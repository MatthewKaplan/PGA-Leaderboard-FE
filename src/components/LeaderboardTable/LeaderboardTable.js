import React, { Component } from "react";
import "./LeaderboardTable.scss";
import Table from "react-bootstrap/Table";

class LeaderboardTable extends Component {
  editPlayerHandler = player => {
    this.props.editPlayerEvent(player);
  };

  deletePlayerHandler = player => {
    this.props.deletePlayerEvent(player);
  };

  render() {
    const { playerArr } = this.props;
    return (
      <div className="leaderboard-table-component">
        <container className="table-head">
          <div className="pga-logo-container">
            <img
              className="pga-logo"
              src={"https://i.imgur.com/ZZt0zk8.png"}
              alt="PGA logo"
            />
          </div>
          <div className="table-header">
            <h3 className="table-title">1ST ANNUAL FRISCO CHAMPIONSHIP</h3>
            <p>FRISCO, TX</p>
          </div>
        </container>
        <Table hover className="leaderboard-table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Score</th>
              <th className="edit-remove">Edit</th>
              <th className="edit-remove">Remove</th>
            </tr>
          </thead>
          <tbody>
            {playerArr.map((player, index) => (
              <tr key={index}>
                <td className="player-name">
                  {player.last_name}, {player.first_name}
                </td>
                <td className="player-score">{player.score}</td>
                <td
                  className="edit-player"
                  data-test="edit-btn"
                  onClick={() => {
                    this.editPlayerHandler(player);
                  }}
                >
                  <img
                    className="edit-player-btn"
                    src={"https://i.imgur.com/sfaD9Mf.png"}
                    alt="Edit player icon button"
                  />
                </td>
                <td
                  className="delete-player"
                  data-test="delete-btn"
                  onClick={() => {
                    this.deletePlayerHandler(player);
                  }}
                >
                  <img
                    className="delete-btn"
                    src={"https://i.imgur.com/ZJ6SKgx.png"}
                    alt="Trash can icon to remove players"
                  />
                </td>
              </tr>
            ))}
          </tbody>
        </Table>
      </div>
    );
  }
}

export default LeaderboardTable;
