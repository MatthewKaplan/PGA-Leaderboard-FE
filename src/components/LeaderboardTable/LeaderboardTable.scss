.leaderboard-table-component {
    width: 98%;
    margin: 10px 0 0 10px;
    overflow: scroll;
    .table td,
    .table th {
        border: none;
        padding: 0;
    }
    .table {
        border-spacing: 5px;
    }
    .table-head {
        height: 115px;
        display: grid;
        grid-template-columns: 1fr 3fr;
        grid-gap: 10px;
        .pga-logo-container {
            img {
                height: 115px;
                margin: 0 auto;
                display: block;
            }
            margin: 0 auto;
        }
        div {
            background: linear-gradient(0deg, rgba(1,15,24,0.96) 20%, rgba(1,15,24,0.76) 47%, rgba(1,14,24,0.66) 100%);
            border: 3px ridge #010F18;
            h3 {
                margin: 30px 0 0 30px;
            }
            p {
                margin-left: 30px;
            }
            height: 115px;
            width: 100%;
            color: white;
        }
    }
    table {
        border-collapse: separate;
    }

    .leaderboard-table {
        border-spacing: 5px;
        color: black;
        th {
            color: white;
        }

        .player-name {
            background: #010F18;
            border: 3px ridge #010F18;
            color: white;
            text-align: center;
            width: 60%;
        }

        .player-score {
            background: #136F9A;
            color: white;
            text-align: center;
            width: 20%;
        }

        .edit-player {
            background: #006400;
            width: 8%;
            padding-top: 3px;
        }

        .delete-player {
            background:#8B0000;
            width: 8%;
            padding-top: 3px;
        }

        .delete-btn,
        .edit-player-btn {
            height: 22px;
            width: 25px;
            margin: 0 auto;
            display: block;
            cursor: pointer;
        }
    }
}