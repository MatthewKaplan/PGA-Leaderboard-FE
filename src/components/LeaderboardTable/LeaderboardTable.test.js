import React from 'react';
import { shallow } from 'enzyme';
import LeaderboardTable from './LeaderboardTable';
import MockData from '../../assets/mockData';

const mockPlayerArr = MockData.mockSinglePlayerData
const mockEditPlayerEvent = jest.fn();
const mockDeletePlayerEvent = jest.fn();

describe('LeaderboardTable', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<LeaderboardTable playerArr={mockPlayerArr} editPlayerEvent={mockEditPlayerEvent} deletePlayerEvent={mockDeletePlayerEvent}/>)
    instance = wrapper.instance()
  })

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should invoke 'editPlayerHandler' with correct params when songs navagation tab is clicked", () => {
    jest.spyOn(instance, "editPlayerHandler");
    wrapper.find("[data-test='edit-btn']").simulate("click");
    expect(instance.editPlayerHandler).toHaveBeenCalledWith(mockPlayerArr[0]);
  });

  it("should invoke 'deletePlayerHandler' with correct params when songs navagation tab is clicked", () => {
    jest.spyOn(instance, "deletePlayerHandler");
    wrapper.find("[data-test='delete-btn']").simulate("click");
    expect(instance.deletePlayerHandler).toHaveBeenCalledWith(mockPlayerArr[0]);
  });
});