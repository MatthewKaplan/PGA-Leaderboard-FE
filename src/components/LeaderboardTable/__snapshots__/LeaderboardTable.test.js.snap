// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardTable should match the snapshot 1`] = `ShallowWrapper {}`;
